### ISSUE TITLE
what is testset.pred_offset used for ?

### ISSUE THREAD

**Issue by oujieww** [zhanghang1989/PyTorch-Encoding#102]
when i use test code for testset ,running to this line:
predicts = [torch.max(output, 1)[1].cpu().numpy() + testset.pred_offset for output in outputs]

but  : 
@property
    def pred_offset(self):
        raise NotImplemented

testset.pred_offset is not implemented, is this important?or not? is there anyone can help me?

**Comment by my462** [comment]
It's used to give the right mask predictions for some certain datasets like VOC2012.

**Comment by oujieww** [comment]
@my462，thank you！so that this None for voc segmentation？

**Comment by my462** [comment]
@oujieww Actually it's zero. When you try to save the VOC 2012's test set's result, you will need to adjust a little bit of author's code.

**Comment by oujieww** [comment]
have u run auther test code？before？i found if u use multi gpus and batch size is not 1，for example i use 4 gpus and batch size 4，only save 1/4 of test set result～～～ so i debug for validation set ，find that only 1/4 output were used for final miou

**Comment by my462** [comment]
If you read the code carefully enough, you will know the test batch size(multi evaluation mode) is the number of your gpus

**Comment by oujieww** [comment]
thank you，i should read the code again XD！

**Comment by oujieww** [comment]
@my462  yes,i change the batchsize to same as number of gpus it is ok now!!!thank you XD! and do i need to set this for training?i use auther's setting meaning auto...

**Comment by ElliotChx** [comment]
@oujieww hi,i met the same as you problem,How did you solve this problem?can you help me?

**Comment by oujieww** [comment]
i think i used 0 or None to ignore this
